pkg/controller/windowsmachineconfig/windowsmachineconfig_controller.go

```diff
@@ -257,201 +166,68 @@ func (r *ReconcileWindowsMachineConfig) Reconcile(request reconcile.Request) (re
-	} else if desired > current {
-		successCount, errs = r.addWorkerNodes(desired - current)
-		newNodeCount = current + successCount
-	} else if desired == current {
-		return current, nil
-	}
-
-	log.V(1).Info("starting tracker reconciliation")
-	if err := r.tracker.Reconcile(); err != nil {
-		errs = append(errs, newReconcileError(wmcapi.TrackerFailureReason, err))
+	if len(ipAddress) == 0 {
+		return reconcile.Result{}, nil
 	}
-	log.V(1).Info("completed tracker reconciliation")
 
-	return newNodeCount, errs
-}
-
-// removeWorkerNode terminates the underlying VM and removes the given vm from the list of VMs
-func (r *ReconcileWindowsMachineConfig) removeWorkerNode(vm types.WindowsVM) ReconcileError {
-	// VM is missing credentials, this can occur if there was a failure initially creating it. We can consider the
-	// actual VM terminated as there is nothing we can do with it.
-	if vm.GetCredentials() == nil || len(vm.GetCredentials().GetInstanceId()) == 0 {
-		r.tracker.deleteWindowsVM(vm)
-		return nil
+	// Get the instanceid associated with the Windows machine.
+	providerID := *machine.Spec.ProviderID
+	if len(providerID) == 0 {
+		return reconcile.Result{}, nil
 	}
-
-	// Terminate the instance via its instance id
-	id := vm.GetCredentials().GetInstanceId()
-	log.V(1).Info("destroying the Windows VM", "ID", id)
-
-	// Delete the Windows VM from cloud provider
-	if err := r.cloudProvider.DestroyWindowsVM(id); err != nil {
-		return newReconcileError(wmcapi.VMTerminationFailureReason,
-			errors.Wrapf(err, "error destroying VM with ID %s", id))
+	// Ex: aws:///us-east-1e/i-078285fdadccb2eaa. We always want the last entry which is the instanceID
+	providerTokens := strings.Split(providerID, "/")
+	instanceID := providerTokens[len(providerTokens)-1]
+	if len(instanceID) == 0 {
+		return reconcile.Result{}, nil
 	}
 
-	// Remove VM from our list of tracked VMs
-	r.tracker.deleteWindowsVM(vm)
-	log.Info("Windows worker has been removed from the cluster", "ID", id)
-	return nil
-}
-
-// removeWorkerNodes removes the required number of Windows VMs from the cluster and returns a bool indicating the
-// success. Returns the actual number of removed nodes and any associated errors.
-func (r *ReconcileWindowsMachineConfig) removeWorkerNodes(count int) (int, []ReconcileError) {
-	var errs []ReconcileError
-	// From the list of Windows VMs choose randomly count number of VMs.
-	for i := 0; i < count; i++ {
-		// Choose of the Windows worker nodes randomly
-		vm := r.tracker.chooseRandomNode()
-		if vm == nil {
-			errs = append(errs, newReconcileError(wmcapi.VMTerminationFailureReason,
-				fmt.Errorf("expected VM and got a nil value")))
-			continue
-		}
-		if err := r.removeWorkerNode(vm); err != nil {
-			errs = append(errs, err)
-		}
+	// Make the Machine a Windows Worker node
+	if err := r.addWorkerNode(ipAddress, instanceID); err != nil {
+		log.Error(err, "Windows VM failed to be configured", "instanceID", instanceID)
```

Comment:
This error will be logged when `return reconcile.Result{}, err`
I do not think adding an extra log message here will add anything